rotkehlchen/globaldb/handler.py

```diff
@@ -1203,6 +1227,144 @@ def get_historical_price_data(source: HistoricalPriceOracle) -> List[Dict[str, A
+                    user_db_cursor.execute(diffs_query)
+                    amount = user_db_cursor.fetchone()
+                    if amount[0] != 0:
+                        cursor.execute(detach_database)
+                        log.error(
+                            f'Assets inserted by the user are used in the database '
+                            f'tables {diff_ids}',
+                        )
+                        return False, f'There are assets that can not be deleted: {str(diff_ids)}'
+            # Check that versions match
+            query = cursor.execute('SELECT value from clean_db.settings WHERE name=="version";')
+            version = query.fetchone()
+            if int(version[0]) != _get_setting_value(cursor, 'version', GLOBAL_DB_VERSION):
+                cursor.execute(detach_database)
+                msg = (
+                    'Failed to restore assets. Global database is not '
+                    'updated to the latest version'
+                )
+                return False, msg
+            # If versions match drop tables
+            cursor.execute('PRAGMA foreign_keys = OFF;')
+            cursor.execute('DELETE FROM assets;')
+            cursor.execute('DELETE FROM ethereum_tokens;')
+            cursor.execute('DELETE FROM underlying_tokens_list;')
+            cursor.execute('DELETE FROM common_asset_details;')
+            # Copy assets
+            cursor.execute('INSERT INTO assets SELECT * FROM clean_db.assets;')
+            cursor.execute('INSERT INTO ethereum_tokens SELECT * FROM clean_db.ethereum_tokens;')
+            cursor.execute('INSERT INTO underlying_tokens_list SELECT * FROM clean_db.underlying_tokens_list;')  # noqa: E501
+            cursor.execute('INSERT INTO common_asset_details SELECT * FROM clean_db.common_asset_details;')  # noqa: E501
+            cursor.execute('PRAGMA foreign_keys = ON;')
+
+            user_db_cursor.execute('PRAGMA foreign_keys = OFF;')
+            user_db_cursor.execute('DELETE FROM assets;')
+            user_db_cursor.execute('PRAGMA foreign_keys = ON;')
+            # Get ids for assets to insert them in the user db
+            cursor.execute('SELECT identifier from assets')
+            ids = cursor.fetchall()
+            ids_proccesed = ", ".join([f'("{id[0]}")' for id in ids])
+            user_db_cursor.execute(f'INSERT INTO assets(identifier) VALUES {ids_proccesed};')
+        except sqlite3.Error as e:
+            connection.rollback()
+            cursor.execute(detach_database)
+            log.error(f'Failed to restore assets in globaldb due to {str(e)}')
+            return False, 'Failed to restore assets. Read logs to get more information.'
+
+        connection.commit()
+        cursor.execute(detach_database)
+
+        return True, ''
+
+    @staticmethod
+    def reset_assets_list() -> Tuple[bool, str]:
+        """Resets all assets to the state in the packaged global db"""
```

Comment:
Mention in the docstring that all custom assets not existing in the packaged global DB remain unaffected.